File: src/test/java/io/binac/leetcode/NQueensTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class NQueensTest {
    private final NQueens.Solution1 solution1 = new NQueens.Solution1();

    @Test
    void test1() {
        List<List<String>> result = solution1.solveNQueens(4);
        List<List<String>> expected = Arrays.asList(
                Arrays.asList(
                        ".Q..",
                        "...Q",
                        "Q...",
                        "..Q."
                ),
                Arrays.asList(
                        "..Q.",
                        "Q...",
                        "...Q",
                        ".Q.."
                )
        );
        assertEquals(expected, result);
    }
}

File: cpp/unittests/NQueensTest.cpp
#include "NQueens.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(NQueens, Solution51_1) {
  const int N = 4;
  const std::vector<std::vector<std::string>> Expected{
      {".Q..",
       "...Q",
       "Q...",
       "..Q."},
      {"..Q.",
       "Q...",
       "...Q",
       ".Q.."}
  };
  EXPECT_EQ(Expected, Solution51_1().solveNQueens(N));
}